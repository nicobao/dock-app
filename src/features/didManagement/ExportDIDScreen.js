import React, {useCallback} from 'react';
import {addTestId} from '../../core/automation-utils';
import {
  BackButton,
  Content,
  Footer,
  Header,
  Input,
  LoadingButton,
  NBox as Box,
  ScreenContainer,
  Typography,
} from '../../design-system';
import {FormControl, Stack} from 'native-base';
import {translate} from '../../locales';
import {CheckCircle} from '../../components/CheckCircleComponent';
import {useDIDManagementHandlers, useExportDIDHandlers} from './didHooks';

export function ExportDIDScreen({form, onChange, formValid, onSubmit}) {
  return (
    <ScreenContainer {...addTestId('ExportDIDScreen')}>
      <Header>
        <BackButton {...addTestId('BackButton')} />
      </Header>
      <Content marginLeft={26} marginRight={26}>
        <Typography variant="h1" marginTop={52}>
          {translate('didManagement.create_password')}
        </Typography>
        <Typography marginTop={3}>
          {translate('didManagement.create_password_desc')}
        </Typography>

        <Box>
          <FormControl>
            <Stack mt={7}>
              <FormControl.Label>
                {translate('didManagement.enter_password')}
              </FormControl.Label>
              <Input
                placeholder={translate('password_input.password')}
                {...addTestId('Password')}
                value={form.password}
                onChangeText={onChange('password')}
                autoCapitalize="none"
                secureTextEntry={true}
              />
            </Stack>
          </FormControl>
        </Box>

        <Box mb={7}>
          <FormControl>
            <Stack mt={7}>
              <FormControl.Label>
                {translate('didManagement.confirm_password')}
              </FormControl.Label>
              <Input
                placeholder={translate('password_input.confirm_password')}
                {...addTestId('ConfirmPassword')}
                value={form.passwordConfirmation}
                onChangeText={onChange('passwordConfirmation')}
                autoCapitalize="none"
                secureTextEntry={true}
              />
            </Stack>
          </FormControl>
        </Box>
        <Stack marginTop={4}>
          <Stack direction="row">
            <CheckCircle checked={form.lengthValidation} />
            <Typography>
              {translate('create_password.include_char_length')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={form.digitsValidation} />
            <Typography>
              {translate('create_password.include_digits')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={form.caseValidation} />
            <Typography>
              {translate('create_password.include_proper_case')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={form.specialCharactersValidation} />
            <Typography>
              {translate('create_password.include_special_characters')}
            </Typography>
          </Stack>
          <Stack direction="row" marginTop={3}>
            <CheckCircle checked={form.passwordMatchValidation} />
            <Typography>
              {translate('create_password.passwords_match')}
            </Typography>
          </Stack>
        </Stack>
      </Content>
      <Footer marginBottom={10} marginLeft={26} marginRight={26}>
        <LoadingButton
          full
          {...addTestId('NextBtn')}
          isDisabled={!formValid}
          onPress={onSubmit}>
          {translate('navigation.next')}
        </LoadingButton>
      </Footer>
    </ScreenContainer>
  );
}
export function ExportDIDScreenContainer({route}) {
  const {didDocumentResolution} = route.params;

  const {form, handleChange, formValid} = useExportDIDHandlers();

  const {onExportDID} = useDIDManagementHandlers();

  const onSubmit = useCallback(async () => {
    await onExportDID({id: didDocumentResolution.id, password: form.password});
  }, [didDocumentResolution.id, form.password, onExportDID]);
  return (
    <ExportDIDScreen
      form={form}
      onChange={handleChange}
      formValid={formValid}
      onSubmit={onSubmit}
    />
  );
}
