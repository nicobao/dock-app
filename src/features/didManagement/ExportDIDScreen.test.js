import React from 'react';
import {shallow} from 'enzyme';
import configureMockStore from 'redux-mock-store';
import {ExportDIDScreen} from './ExportDIDScreen';

const mockStore = configureMockStore();

describe('ExportDIDScreen', () => {
  it('should render correctly when form valid', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      password: '',
      passwordConfirmation: '',
      _errors: {},
      _hasError: false,
    };
    const wrapper = shallow(
      <ExportDIDScreen
        form={form}
        onChange={jest.fn()}
        formValid={true}
        onSubmit={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
  it('should render correctly when form invalid', () => {
    const initialState = {
      // placeholder for redux store
    };

    const form = {
      password: '',
      passwordConfirmation: '',
      _errors: {},
      _hasError: false,
    };
    const wrapper = shallow(
      <ExportDIDScreen
        form={form}
        onChange={jest.fn()}
        formValid={false}
        onSubmit={jest.fn()}
      />,
      {
        context: {store: mockStore(initialState)},
      },
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
