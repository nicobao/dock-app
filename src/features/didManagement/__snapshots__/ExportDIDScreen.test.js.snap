// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportDIDScreen should render correctly when form invalid 1`] = `
<RNCSafeAreaView
  accessibilityLabel="ExportDIDScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="ExportDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <BackButton
          accessibilityLabel="BackButton"
          testID="BackButton"
        />
      </Styled(View)>
      <Styled(ScrollView)
        marginLeft={26}
        marginRight={26}
      >
        <Styled(Component)
          marginTop={52}
          variant="h1"
        >
          Create a password
        </Styled(Component)>
        <Styled(Component)
          marginTop={3}
        >
          This will be used to import your DID to a new device
        </Styled(Component)>
        <Memo(ForwardRef(Box))>
          <Memo(ForwardRef(FormControl))>
            <Memo(ForwardRef(Stack))
              mt={7}
            >
              <Memo(ForwardRef(FormControlLabel))>
                Enter Password
              </Memo(ForwardRef(FormControlLabel))>
              <Memo(ForwardRef(Input))
                accessibilityLabel="Password"
                autoCapitalize="none"
                placeholder="Password"
                secureTextEntry={true}
                testID="Password"
                value=""
              />
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(FormControl))>
        </Memo(ForwardRef(Box))>
        <Memo(ForwardRef(Box))
          mb={7}
        >
          <Memo(ForwardRef(FormControl))>
            <Memo(ForwardRef(Stack))
              mt={7}
            >
              <Memo(ForwardRef(FormControlLabel))>
                Confirm Password
              </Memo(ForwardRef(FormControlLabel))>
              <Memo(ForwardRef(Input))
                accessibilityLabel="ConfirmPassword"
                autoCapitalize="none"
                placeholder="Confirm Password"
                secureTextEntry={true}
                testID="ConfirmPassword"
                value=""
              />
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(FormControl))>
        </Memo(ForwardRef(Box))>
        <Memo(ForwardRef(Stack))
          marginTop={4}
        >
          <Memo(ForwardRef(Stack))
            direction="row"
          >
            <CheckCircle />
            <Styled(Component)>
              Must include at least 8 characters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Must include digits
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Must include uppercase and lowercase letters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Must include special characters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Password confirmation match
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(Stack))>
      </Styled(ScrollView)>
      <Styled(Component)
        marginBottom={10}
        marginLeft={26}
        marginRight={26}
      >
        <LoadingButton
          accessibilityLabel="NextBtn"
          full={true}
          isDisabled={true}
          onPress={[MockFunction]}
          testID="NextBtn"
        >
          Next
        </LoadingButton>
      </Styled(Component)>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;

exports[`ExportDIDScreen should render correctly when form valid 1`] = `
<RNCSafeAreaView
  accessibilityLabel="ExportDIDScreen"
  style={
    Object {
      "backgroundColor": "#141417",
      "flex": 1,
    }
  }
  testID="ExportDIDScreen"
>
  <AppGlobalHeader />
  <Memo(ForwardRef(Box))
    flex={1}
  >
    <Memo(ForwardRef(Box))
      flex={1}
    >
      <Styled(View)>
        <BackButton
          accessibilityLabel="BackButton"
          testID="BackButton"
        />
      </Styled(View)>
      <Styled(ScrollView)
        marginLeft={26}
        marginRight={26}
      >
        <Styled(Component)
          marginTop={52}
          variant="h1"
        >
          Create a password
        </Styled(Component)>
        <Styled(Component)
          marginTop={3}
        >
          This will be used to import your DID to a new device
        </Styled(Component)>
        <Memo(ForwardRef(Box))>
          <Memo(ForwardRef(FormControl))>
            <Memo(ForwardRef(Stack))
              mt={7}
            >
              <Memo(ForwardRef(FormControlLabel))>
                Enter Password
              </Memo(ForwardRef(FormControlLabel))>
              <Memo(ForwardRef(Input))
                accessibilityLabel="Password"
                autoCapitalize="none"
                placeholder="Password"
                secureTextEntry={true}
                testID="Password"
                value=""
              />
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(FormControl))>
        </Memo(ForwardRef(Box))>
        <Memo(ForwardRef(Box))
          mb={7}
        >
          <Memo(ForwardRef(FormControl))>
            <Memo(ForwardRef(Stack))
              mt={7}
            >
              <Memo(ForwardRef(FormControlLabel))>
                Confirm Password
              </Memo(ForwardRef(FormControlLabel))>
              <Memo(ForwardRef(Input))
                accessibilityLabel="ConfirmPassword"
                autoCapitalize="none"
                placeholder="Confirm Password"
                secureTextEntry={true}
                testID="ConfirmPassword"
                value=""
              />
            </Memo(ForwardRef(Stack))>
          </Memo(ForwardRef(FormControl))>
        </Memo(ForwardRef(Box))>
        <Memo(ForwardRef(Stack))
          marginTop={4}
        >
          <Memo(ForwardRef(Stack))
            direction="row"
          >
            <CheckCircle />
            <Styled(Component)>
              Must include at least 8 characters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Must include digits
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Must include uppercase and lowercase letters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Must include special characters
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
          <Memo(ForwardRef(Stack))
            direction="row"
            marginTop={3}
          >
            <CheckCircle />
            <Styled(Component)>
              Password confirmation match
            </Styled(Component)>
          </Memo(ForwardRef(Stack))>
        </Memo(ForwardRef(Stack))>
      </Styled(ScrollView)>
      <Styled(Component)
        marginBottom={10}
        marginLeft={26}
        marginRight={26}
      >
        <LoadingButton
          accessibilityLabel="NextBtn"
          full={true}
          isDisabled={false}
          onPress={[MockFunction]}
          testID="NextBtn"
        >
          Next
        </LoadingButton>
      </Styled(Component)>
    </Memo(ForwardRef(Box))>
  </Memo(ForwardRef(Box))>
</RNCSafeAreaView>
`;
